### PR TITLE
learner need load merge target

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

A learner should load merge target if another overlapped region wants to apply snapshot or create a peer.
So add a StatusCmdType to handle this.

For https://github.com/tikv/tikv/pull/6598

### CODE REVIEW

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -22,6 +22,7 @@ message RaftMessage {
     bytes end_key = 8;
     // If it has value, to_peer should be removed if merge is never going to complete.
     metapb.Region merge_target = 9;
+    MergeMsg merge = 10;
```
Comment (BusyJay) [inline proto/raft_serverpb.proto]: How about using StatusCommand in RaftCmdRequest?

File: proto/raft_cmdpb.proto [hunk 2]
```diff
@@ -258,6 +258,8 @@ enum StatusCmdType {
     InvalidStatus = 0;
     RegionLeader = 1;
     RegionDetail = 2;
+    RegionMergeTarget = 3;
+    RegionWakeUp = 4;
```
Comment (Connor1996) [inline proto/raft_cmdpb.proto]: what's used for

Comment (gengliqi) [reply proto/raft_cmdpb.proto]: RegionMergeTarget used for learner peer load merge target.
RegionWakeUp used for waking up the hibernate region.

Comment (BusyJay) [reply proto/raft_cmdpb.proto]: How about just keeping `RegionWakeUp`?

**Pull request merged**